src/include/simeng/arch/Architecture.hh

```diff
@@ -14,6 +14,12 @@ using MacroOp = std::vector<std::shared_ptr<Instruction>>;
 
 namespace arch {
 
+/** Modes. Assume only has 32-bit and 64-bit. */
+enum arch_mode {
```

Comment:
Please use the enum class to establish an explicit type, and parent it on uint8_t to reduce size usage.
Also please capitalise arch_mode. 

You can static_cast the enum values to be used as an index or anything else.